cli/cmd/completion.go

```diff
@@ -0,0 +1,294 @@
+		if eval "[[ \"\${w}\" = *.$1 || -d \"\${w}\" ]]"; then
+			qw="$(__lokoctl_quote "${w}")"
+			if [ -d "${w}" ]; then
+				COMPREPLY+=("${qw}/")
+			else
+				COMPREPLY+=("${qw}")
+			fi
+		fi
+	done
+}
+__lokoctl_quote() {
+	if [[ $1 == \'* || $1 == \"* ]]; then
+		# Leave out first character
+		printf %q "${1:1}"
+	else
+		printf %q "$1"
+	fi
+}
+autoload -U +X bashcompinit && bashcompinit
+# use word boundary patterns for BSD or GNU sed
+LWORD='[[:<:]]'
+RWORD='[[:>:]]'
+if sed --help 2>&1 | grep -q 'GNU\|BusyBox'; then
+	LWORD='\<'
+	RWORD='\>'
+fi
+__lokoctl_convert_bash_to_zsh() {
+	sed \
+	-e 's/declare -F/whence -w/' \
+	-e 's/_get_comp_words_by_ref "\$@"/_get_comp_words_by_ref "\$*"/' \
+	-e 's/local \([a-zA-Z0-9_]*\)=/local \1; \1=/' \
+	-e 's/flags+=("\(--.*\)=")/flags+=("\1"); two_word_flags+=("\1")/' \
+	-e 's/must_have_one_flag+=("\(--.*\)=")/must_have_one_flag+=("\1")/' \
+	-e "s/${LWORD}_filedir${RWORD}/__lokoctl_filedir/g" \
+	-e "s/${LWORD}_get_comp_words_by_ref${RWORD}/__lokoctl_get_comp_words_by_ref/g" \
+	-e "s/${LWORD}__ltrim_colon_completions${RWORD}/__lokoctl_ltrim_colon_completions/g" \
+	-e "s/${LWORD}compgen${RWORD}/__lokoctl_compgen/g" \
+	-e "s/${LWORD}compopt${RWORD}/__lokoctl_compopt/g" \
+	-e "s/${LWORD}declare${RWORD}/builtin declare/g" \
+	-e "s/\\\$(type${RWORD}/\$(__lokoctl_type/g" \
+	-e 's/aliashash\["\(.\{1,\}\)"\]/aliashash[\1]/g' \
+	-e 's/FUNCNAME/funcstack/g' \
+	<<'BASH_COMPLETION_EOF'
+`
+
+const zshTail = `
+BASH_COMPLETION_EOF
+}
+__lokoctl_bash_source <(__lokoctl_convert_bash_to_zsh)
+`
+
+func newCompletionCmd(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "completion",
+		Short:             "Generate autocompletion script for the specified shell",
+		Long:              completionDesc,
+		Example:           completionExample,
+		Args:              noArgs,
+		ValidArgsFunction: noCompletions, // Disable file completion.
+	}
+
+	bash := &cobra.Command{
+		Use:                   "bash",
+		Short:                 "Generate autocompletion script for bash",
+		Long:                  bashCompDesc,
+		Args:                  noArgs,
+		DisableFlagsInUseLine: true,
+		ValidArgsFunction:     noCompletions,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runCompletionBash(out, cmd)
+		},
+	}
+
+	zsh := &cobra.Command{
+		Use:                   "zsh",
+		Short:                 "Generate autocompletion script for zsh",
+		Long:                  zshCompDesc,
+		Args:                  noArgs,
+		DisableFlagsInUseLine: true,
+		ValidArgsFunction:     noCompletions,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runCompletionZsh(out, cmd)
+		},
+	}
+
+	cmd.AddCommand(bash, zsh)
+
+	return cmd
+}
+
+func runCompletionBash(out io.Writer, cmd *cobra.Command) error {
+	return cmd.Root().GenBashCompletion(out)
+}
+
+func runCompletionZsh(out io.Writer, cmd *cobra.Command) error {
+	_, err := out.Write([]byte(zshInitialization))
+	if err != nil {
```

Comment:
Please consolidate lines which only assigns `err` with `if` statement lines.
```suggestion
	if _, err := out.Write([]byte(zshInitialization)); err != nil {
```